chapter/8/big-data.md

```diff
@@ -1,31 +1,46 @@
 ---
 layout: page
 title:  "Large Scale Parallel Data Processing"
-by: "Jingjing and Abhilash"
+by: "Jingjing and Abhilash Mysore Somashekar"
 ---
 ## Introduction
-The growth of Internet has generated the so-called big data(terabytes or petabytes). It is not possible to fit them into a single machine or process them with one single program. Often the computation has to be done fast enough to provide practical services. A common approach taken by tech giants like Google, Yahoo, Facebook is to process big data across clusters of commodity machines. Many of the computations are conceptually straightforward, and Google proposed the MapReduce model to abstract the logic and proved to be simple and powerful. From then on, the idea inspired lots of other programming models. In this chapter, we will present how programming models evolve over time, why their execution engines are designed in certain ways, and underlying ecosystem that supports each developing thread.
+The growth of Internet has generated the so-called big data(terabytes or petabytes). It is not possible to fit them into a single machine or process them with one single program. Often the computation has to be done fast enough to provide practical services. A common approach taken by tech giants like Google, Yahoo, Facebook is to process big data across clusters of commodity machines. Many of the computations are conceptually straightforward, and Google proposed the MapReduce framework, which separates the programming logic and underlying execution details(data distribution, fault tolerance and scheduling). The model has been proved to be simple and powerful, and from then on, the idea inspired many other programming models.
```

Comment:
It might be useful to clarify that it's not just about the division of logic and execution, it's also the division between things that can be executed in parallel and things that need coordination between nodes: ie. the division between map and reduce phases, where reduce requires coordination amongst nodes and map phases are usually, or rather by definition, 'embarrassingly parallel.'